builder/qemu/step_prepare_efivars.go

```diff
@@ -0,0 +1,71 @@
+package qemu
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+)
+
+// stepPrepareEfivars copies the EFIVars file to the output, so we can boot
+// and use it as a RW flash drive
+type stepPrepareEfivars struct {
+	EFIEnabled  bool
+	OutputDir   string
+	EFIVarsPath string
+}
+
+const efivarStateKey string = "EFI_VARS_FILE_PATH"
+
+func (prepEfi *stepPrepareEfivars) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packersdk.Ui)
+
+	if !prepEfi.EFIEnabled {
+		return multistep.ActionContinue
+	}
+
+	efiVarsOutput := fmt.Sprintf("%s/efivars.fd", prepEfi.OutputDir)
+	outFile, err := os.OpenFile(efiVarsOutput, os.O_CREATE|os.O_WRONLY, 0660)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed to create local efivars file: %s", err)
```

Comment:
Should we include the name of the file that was trying to be created?

Reply:
We could but not sure this'll help the user knowing that a temp file in output failed to be created, as it's not a path they'll be using for other purposes. The only way this can fail is if the output directory doesn't exist (which will already be reported beforehand when creating the output image), or if for some reason the FS fails, which isn't something a user can do much about I think.
That being said, I can add the path to the message, it can't hurt either.